vault/core.go

```diff
@@ -4051,3 +4051,30 @@ func (c *Core) GetRaftAutopilotState(ctx context.Context) (*raft.AutopilotState,
 
 	return raftBackend.GetAutopilotServerState(ctx)
 }
+
+func (c *Core) initializeUnsealRecovery(ctx context.Context, recoveryKey, barrierKey []byte) error {
+	wrapper := aeadwrapper.NewShamirWrapper()
+	if err := wrapper.SetAesGcmKeyBytes(recoveryKey); err != nil {
+		return fmt.Errorf("failed to store recover unseal keys: %v", err)
+	}
+	recoverySeal := NewRecoverySeal(&vaultseal.Access{
+		Wrapper: wrapper,
+	})
+	recoverySeal.SetCore(c)
+	if err := recoverySeal.SetStoredKeys(ctx, [][]byte{barrierKey}); err != nil {
+		return fmt.Errorf("failed to store recover unseal keys: %v", err)
+	}
+	return nil
+}
+
+func (c *Core) UnsealRecoveryEnabled(ctx context.Context) (bool, error) {
+	// TODO: Return false on ENT
```

Comment:
Could you just add a `constants.IsEnterprise || (...)` here?

Reply:
Oh sweet, TiL.